src/Tribe/Utils/Post_Root_Pool.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+
+class Tribe__Utils__Post_Root_Pool {
+
+	/**
+	 * @var string
+	 */
+	protected $pool_transient_name = 'tribe_ticket_prefix_pool';
+
+	/**
+	 * @var array|bool
+	 */
+	protected static $prefix_pool = false;
+
+	/**
+	 * @var string
+	 */
+	protected $root_separator = '-';
+
+	/**
+	 * @var array
+	 */
+	protected $postfix = 1;
+
+	/**
+	 * Generates a unique root for a post using its post_name.
+	 *
+	 * @param WP_Post $post
+	 *
+	 * @return string
+	 */
+	public function generate_unique_root( WP_Post $post ) {
```

Comment:
The post root pool is meant to be generic and agnostic of where it's used.
The context calling it, in this case the `Tribe__Tickets_Plus__Meta__Unique_ID` class, should take care, as it does, to generate new roots only when needed.
Updating the doc block just in case.